### ISSUE TITLE
borderLayout templates with inline styles violate Content Security Policy

### ISSUE THREAD

**Issue by farrago** [ceolter/ag-grid#1093]
### Summary
Inline styles in borderLayout's templates break CSP.  Moving the styles to the css appears to resolve the issue.

### Details
The templates for [borderLayout (L43-75)](https://github.com/ceolter/ag-grid/blob/master/src/ts/layout/borderLayout.ts#L43) include inline styles e.g.:
```javascript
template = 
  '<div style="height: 100%; overflow: auto; position: relative;">' +
  ...
```

If [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/Security/CSP) (CSP) is enabled, these inline styles cannot be applied and cause errors.  e.g. in chrome:

>[Report Only] Refused to apply inline style because it violates the following Content Security Policy directive: "default-src 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-xHgdoELGuMXNdMeJ4PQkbzZpH1rlmZwpY3b56d2ZvpI='), or a nonce ('nonce-...') is required to enable inline execution. Note also that 'style-src' was not explicitly set, so 'default-src' is used as a fallback.

I've tried moving these styles from inline in the templates to the CSS, and that appears to work fine visually and resolves the CSP errors.

 **Is this something that could be done in the official release?**

### Setup
Browser: Chrome `52.0.2743.116 m` (up to date), plus same result in Firefox, Edge, etc.
OS: Windows 10
Server: NodeJS 4.4.3
CSP Headers: `content-security-policy-report-only: default-src 'self'; report-uri /api/v0/csp-report` (This reports but doesn't prevent violations of the policy while debugging rules).

### Notes
As per the error reported by Chrome, this could theoretically be worked around by using `'unsafe-inline'` but this is, as named, unsafe.  It could also be worked around with a hash for each case but that isn't well supported by browsers yet, and there are a lot of different cases that would need a hash for each one. Nonces are even harder to use with library code.

**Comment by ceolter** [comment]
moving the styles from the template into css files makes sense, and can be done here. however, there are other parts of the app that it cannot be easily done, eg the row positioning is done using pixels and set as inline styles. can you advise why this is ok to do, but not the styles in border layout?

**Comment by farrago** [comment]
An excellent question, though not quite sure of the answer! 

The actual line of the CSP error is the `tempDiv.innerHTML = template;` in:

```javascript
	    /**
	     * loads the template and returns it as an element. makes up for no simple way in
	     * the dom api to load html directly, eg we cannot do this: document.createElement(template)
	     */
	    Utils.loadTemplate = function (template) {
	        var tempDiv = document.createElement("div");
	        tempDiv.innerHTML = template;
	        return tempDiv.firstChild;
	    };
```

Setting the `innerHTML` with text including a `style="..."` presumably counts as an inline style.

From a quick search through the code, it seems that everywhere else sets the `HTMLElement.style` property, which presumably doesn't count as an inline style (i.e. it isn't parsed from HTML text but is set directly)?

That would seem to be the most likely reason, but perhaps someone else can confirm or deny this?

**Comment by ceolter** [comment]
ok thanks. i've added this to my list. will get done over next two weeks approx.

**Comment by farrago** [comment]
Thanks, much appreciated!

**Comment by ceolter** [comment]
this is now done. will be in next release.

**Comment by farrago** [comment]
I've just tested the new release in the setup that was causing the original issue and I can confirm that the issue is resolved in 5.4.0.

Thanks for your quick response and release.
